x/auth/types/account.go

```diff
@@ -301,6 +336,41 @@ type AccountI interface {
 
 	// Ensure that account implements stringer
 	String() string
+
+	MarshalX() ([]byte, error)
+}
+
+func MarshalAccountX(cdc codec.BinaryMarshaler, acc AccountI) ([]byte, error) {
+	if bacc, ok := acc.(*BaseAccount); ok && bacc.MultisigPubKey == nil {
+		return acc.MarshalX()
+	} else if macc, ok := acc.(*ModuleAccount); ok && macc.MultisigPubKey == nil {
+		return acc.MarshalX()
+	} else {
+		return cdc.MarshalInterface(acc)
+	}
+}
+
+func UnmarshalAccountX(cdc codec.BinaryMarshaler, bz []byte) (AccountI, error) {
+	sigLen := len(BaseAccountSig)
+	if len(bz) < sigLen {
+		return nil, fmt.Errorf("invalid data")
+	}
+	if bytes.Equal(bz[:sigLen], BaseAccountSig) {
+		acc := &BaseAccount{}
+		if err := acc.Unmarshal(bz[sigLen:]); err != nil {
+			return nil, err
+		}
+		return acc, nil
+	} else if bytes.Equal(bz[:sigLen], ModuleAccountSig) {
```

Comment:
There is no contract that `BaseAccountSig ` and `ModuleAccountSig ` are the same length.
This is likely to lead to mistakes later.

Reply:
I agree. Will fix it with a separate check-in.